lib/services/push_notifications/push_notifications.dart

```diff
@@ -34,6 +34,10 @@ class PushNotificationsService {
     OneSignal.shared.setNotificationReceivedHandler(_onNotificationReceived);
     OneSignal.shared.setNotificationOpenedHandler(_onNotificationOpened);
     OneSignal.shared.setSubscriptionObserver(_onSubscriptionChanged);
+    var isSubscribed = await this.isSubscribedToPushNotifications();
+    if (isSubscribed) {
```

Comment:
Isn't this an issue with new users?

They will ofc not be subscribed to notifications and if its never requested, they never will.

Reply:
Yes, turned out the fix was a bit more complicated due to an open issue in OneSignal, Ive created a workaround ( see comments in method) which can be deprecated once they release a new SDK version.